eos/eos.hpp

```diff
@@ -222,11 +220,10 @@ template <typename T> class ShiftedEOS {
 public:
   // move semantics ensures dynamic memory comes along for the ride
   ShiftedEOS(T &&t, const Real shift) : t_(std::forward<T>(t)), shift_(shift) {}
-  PORTABLE_FUNCTION ShiftedEOS() = default;
-  PORTABLE_FUNCTION ShiftedEOS &operator=(ShiftedEOS &&) = default;
-  PORTABLE_FUNCTION ShiftedEOS &operator=(const ShiftedEOS &) = default;
-  PORTABLE_FUNCTION ShiftedEOS(const ShiftedEOS &) = default;
-  // PORTABLE_FUNCTION ~ShiftedEOS() = default;
+  ShiftedEOS() = default;
```

Comment:
If we're removing the device markings (which I think we'll regret when we try to remove Cuda warnings in earnest), why not remove all of the `=default` code as well?

Reply:
The device markings are apparently automatic for defaulted functions. This compiles and runs on device with zero warnings.

As for removing the defaulted functions completely.... yes, we probably should... I'm not sure which of those are deleted if I don't add them. Probably only the default constructor, which we need. I can give that a try.

Reply:
ooooohhhhh, this removed the warnings!! That's great! Please try that out, code removal is good.

Reply:
Done. We can remove everything except the default constructor, which needs to be explicitly defaulted so it doesn't get deleted. (And we need it for kokkos views.)